common/types/ballot.go

```diff
@@ -38,10 +38,12 @@ func (id *BallotID) DecodeScale(d *scale.Decoder) (int, error) {
 
 // Ballot contains the smesher's signed vote on the mesh history.
 type Ballot struct {
-	// the actual votes on the mesh history
+	// InnerBallot is a signed part of the ballot.
 	InnerBallot
 	// smesher's signature on InnerBallot
 	Signature []byte
+	// Votes field is not signed.
+	Votes Votes
```

Comment:
InnerBallot is signed, the same as before

Reply:
yes. but i am talking about ID calculation.

https://github.com/spacemeshos/go-spacemesh/blob/ce3b65c90e73d4d69be78e0eec2b573427cf0068/common/types/ballot.go#L187

the code hashes the whole ballot, including `ballot.Votes`. we should hash data with `ballot.Votes` cleared so that ID doesn't change even tho the votes change.

Reply:
thank i forgot that change was made. it should encode only inner and signature for that purpose